### PR TITLE
fix: The form item margin should appear below extraDom.

### LINKED ISSUE #50026: form 表单项有校验信息时  extra 出现非预期的间距
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/edit/react-e7cstv?file=index.css,demo.tsx)

### Steps to reproduce

使用官方文档中的注册新用户表单case，通过design token设置 itemMarginBottom 或用自定义样式覆盖 .ant-form-item的margin-bottom, 超出默认的24px，当表单项验证错误，显示校验文本时，表单项的 extra 距离校验文本多出了 item margin-bottom的间距，这应该是不符合预期的，原因是与校验文本同级的div中多出了一个div，内联设置高度为 item的 margin-bottom

### What is expected?

extra与校验文本元素间不应该有item margin-bottom 的间距

### What is actually happening?

extra与校验文本元素间出现了 item margin-bottom 的间距

| Environment | Info |
| --- | --- |
| antd | 5.19.3 |
| React | react |
| System | Mac |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by hongzzz**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

close https://github.com/ant-design/ant-design/issues/50026

### 💡 Background and Solution

Removed the div used to expand the error height, wrapped error and extra with a div, and set the minimum height for that div.

### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Form:  Removed the div used to expand the error height, wrapped error and extra with a div, and set the minimum height for that div.      |
| 🇨🇳 Chinese |     Form: 移除了用于撑开error高度的 div，将 errorDom 和 extraDom 用一个 div 包裹，并为该 div 设置了最小高度。      |

### CODE REVIEW

**Comment by hongzzz** [comment]
> ## 👁 Visual Regression Report for PR #50411 Failed ❌
> > **🎯 Target branch:** master ([aa60e53](https://github.com/ant-design/ant-design/commit/aa60e53d4c9acb3f150fcfa36315155c21e93a82))
> > 📖 [View Full Report ↗︎](https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-50411/visualRegressionReport/report.html)
> 
> Expected (Branch master)	Actual (Current PR)	Diff
>  form-disabled-input-debug.compact.png	 form-disabled-input-debug.compact.png	
>  form-disabled-input-debug.compact.png	 form-disabled-input-debug.compact.css-var.png	
>  form-disabled-input-debug.dark.png	 form-disabled-input-debug.dark.png	
>  form-disabled-input-debug.dark.png	 form-disabled-input-debug.dark.css-var.png	
>  form-disabled-input-debug.default.png	 form-disabled-input-debug.default.png	
>  form-disabled-input-debug.default.png	 form-disabled-input-debug.default.css-var.png	
>  form-validate-static.compact.png	 form-validate-static.compact.png	
>  form-validate-static.compact.png	 form-validate-static.compact.css-var.png	
>  form-validate-static.dark.png	 form-validate-static.dark.png	
>  form-validate-static.dark.png	 form-validate-static.dark.css-var.png	
> Check [Full Report](https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-50411/visualRegressionReport/report.html) for details
> 
> If you think the visual diff is acceptable, please check:
> 
> * [ ]  Visual diff is acceptable

I'll check it tonight.

File: components/form/FormItemInput.tsx [hunk 1]
```diff
@@ -109,8 +106,12 @@ const FormItemInput: React.FC<FormItemInputProps & FormItemInputMiscProps> = (pr
     ) : (
       <>
         {inputDom}
-        {errorListDom}
-        {extraDom}
+        <div
+          style={{ display: 'flex', flexDirection: 'column', minHeight: marginBottom || 'unset' }}
```
Comment (yoyo837) [inline components/form/FormItemInput.tsx:110]: 这个能cssinjs不写行内吗？

**Comment by li-jia-nan** [comment]
这是个 breaking change 吧

**Comment by hongzzz** [comment]
> 这是个 breaking change 吧

可能算是，因为改动了结构

**Comment by li-jia-nan** [comment]
> 可能算是，因为改动了结构； 这里还漏了一种 _internalItemRender 的场景，这两天我再看下怎么解

breaking change 提交到 feature 分支吧

File: components/form/FormItemInput.tsx [hunk 2]
```diff
@@ -71,22 +71,20 @@ const FormItemInput: React.FC<FormItemInputProps & FormItemInputMiscProps> = (pr
...
-            warnings={warnings}
-            help={help}
-            helpStatus={status}
-            className={`${baseClassName}-explain-connected`}
-            onVisibleChanged={onErrorVisibleChanged}
-          />
-        </FormItemPrefixContext.Provider>
-        {!!marginBottom && <div style={{ width: 0, height: marginBottom }} />}
-      </div>
+    errors.length || warnings.length || hasHelp ? (
```
Comment (li-jia-nan) [inline components/form/FormItemInput.tsx:76]:
```suggestion
    errors.length > 0 || warnings.length > 0 || hasHelp ? (
```

File: components/auto-complete/__tests__/__snapshots__/demo.test.tsx.snap [hunk 3]
```diff
@@ -383,6 +386,9 @@ exports[`renders components/auto-complete/demo/form-debug.tsx correctly 1`] = `
             </div>
           </div>
         </div>
+        <div
+          class="ant-form-item-additional"
+        />
```
Comment (afc163) [inline components/auto-complete/__tests__/__snapshots__/demo.test.tsx.snap:391]: 为啥会多一个空 div？

Comment (afc163) [reply components/auto-complete/__tests__/__snapshots__/demo.test.tsx.snap:391]: 最好判断一下，避免对 snap 不必要的影响。

**Comment by yoyo837** [comment]
再同步一次源分支吧

**Comment by hongzzz** [comment]
已同步

**Pull request closed without merging**